tools/flaky-test-reporter/ghutil/ghutil.go

```diff
@@ -0,0 +1,327 @@
+	// IssueCloseState is the state of closed github issue
+	IssueCloseState IssueStateEnum = "close"
+	// IssueAllState can be used as filter for querying genericList including closed genericList
+	IssueAllState  IssueStateEnum = "all"
+)
+
+// IssueStateEnum represents different states of Github Issues
+type IssueStateEnum string
+
+// client for github authentication
+// without explicitly authenticate this client can still be used for anonymous operations
+var client *github.Client
+var ctx = context.Background()
+
+// Authenticate explicitly authenticate github with giving token
+func Authenticate(tokenFilePath *string) error {
+	b, err := ioutil.ReadFile(*tokenFilePath)
+	if err != nil {
+		return err
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: strings.TrimSpace(string(b))},
+	)
+	client = github.NewClient(oauth2.NewClient(ctx, ts))
+	return nil
+}
+
+// GetCurrentUser gets authenticated user
+func GetCurrentUser() (*github.User, error) {
+	var res *github.User
+	_, err := retry(
+		fmt.Sprintf("getting current user"),
+		maxRetryCount,
+		func() (*github.Response, error) {
+			var resp *github.Response
+			var err error
+			res, resp, err = client.Users.Get(ctx, "")
+			return resp, err
+		},
+	)
+	return res, err
+}
+
+// ListIssuesByRepo lists genericList within given repo,
+// filters by labels if labels provided
+func ListIssuesByRepo(org, repo string, labels []string) ([]*github.Issue, error) {
+	issueListOptions := github.IssueListByRepoOptions{
+		State:  string(IssueAllState),
+	}
+	if len(labels) > 0 {
+		issueListOptions.Labels = labels
+	}
+	
+	var res []*github.Issue
+	options := &github.ListOptions{}
+	genericList, err := depaginate(
+		fmt.Sprintf("listing genericList with label '%v'", labels),
+		maxRetryCount,
+		options,
+		func() ([]interface{}, *github.Response, error) {
+			page, resp, err := client.Issues.ListByRepo(ctx, org, repo, &issueListOptions)
+			var interfaceList []interface{}
+			if nil == err {
+				for _, issue := range page {
+					interfaceList = append(interfaceList, issue)
+				}
+			}
+			return interfaceList, resp, err
+		},
+	)
+	for _, issue := range genericList {
+		res = append(res, issue.(*github.Issue))
+	}
+	return res, err
+}
+
+// CreateIssue creates issue
+func CreateIssue(org, repo, title, body string) (*github.Issue, error) {
+	issue := &github.IssueRequest{
+		Title: &title,
+		Body:  &body,
+	}
+
+	var res *github.Issue
+	_, err := retry(
+		fmt.Sprintf("creating issue '%s %s' '%s'", org, repo, title),
+		maxRetryCount,
+		func() (*github.Response, error) {
+			var resp *github.Response
+			var err error
+			res, resp, err = client.Issues.Create(ctx, org, repo, issue)
+			return resp, err
+		},
+	)
+	return res, err
+}
+
+// CloseIssue closes issue
+func CloseIssue(org, repo string, number int) error {
+	return updateIssueState(org, repo, IssueCloseState, number)
+}
+
+// ReopenIssue reopen issue
+func ReopenIssue(org, repo string, number int) error {
```

Comment:
Should you check if the issue is closed first? If it's open, would reopen fail?

s/number/issueNumber/

Reply:
reopen doesn't fail if it's open
done renaming number to issueNumber